src/poetry/factory.py

```diff
@@ -126,10 +134,15 @@ def create_config(cls, io: Optional["IO"] = None) -> Config:
         return config
 
     @classmethod
-    def configure_sources(
-        cls, poetry: Poetry, sources: List[Dict[str, str]], config: Config, io: "IO"
-    ) -> None:
-        for source in sources:
+    def create_pool(cls, config: "Config", io: Optional["IO"] = None) -> "Pool":
```

Comment:
These changes look good. I'd love to see a more detailed commit message, and maybe the addition of `PluginCommandMixin` broken out into a separate commit.